cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDataset.java

```diff
@@ -129,23 +198,49 @@ public BusinessMetadataRecord getBusinessMetadata(Id.NamespacedId targetId, Stri
   }
 
   /**
+   * Retrieves all the properties for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the {@link Id.NamespacedId} for which properties are to be retrieved
+   * @return the properties of the specified {@link Id.NamespacedId}
+   */
+  public Map<String, String> getProperties(Id.NamespacedId targetId) {
```

Comment:
We should standardize between the return type of ``getProperties(Id.NamespacedId targetId)`` and ``getProperty(Id.NamespacedId targetId, String key)``.

They should either respectively return:
1.  ``Map<String, String>`` and ``Map.Entry<String, String>``; or
2. ``List<BusinessMetadataRecord>`` and ``BusinessMetadataRecord``.

@poornachandra @henry-cask thoughts?